src/bech32.cpp

```diff
@@ -147,11 +148,15 @@ namespace bech32
 std::string Encode(const std::string& hrp, const data& values) {
     data checksum = CreateChecksum(hrp, values);
     data combined = Cat(values, checksum);
-    std::string ret = hrp + '1';
-    ret.reserve(ret.size() + combined.size());
+    size_t hrp_size = hrp.size();
+    char ret[hrp_size + combined.size() + 2];
+    memcpy(ret, hrp.c_str(), hrp_size);
+    char* ptr = ret + hrp_size;
+    *(ptr++) = '1';
     for (auto c : combined) {
-        ret += CHARSET[c];
+        *(ptr++) = CHARSET[c];
```

Comment:
How about switching to `++ptr`?

Reply:
Why? (And that would change the behavior to setting the next value rather than the current and iterating.)

Reply:
`++i is preferred over i++.`
https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md

You can make the behavior fit by shifting the increment to the later operation in each case.

Reply:
In this case, I think `$(ptr++)` is a better alternative, as it is more intuitive. Update the value and then move to the next position. Doing `++i` here semantically means the pointer points to the last byte that was set, rather than the next available byte.

Reply:
The nice thing about `++ptr`, and the reason it is preferred AFAIK, is that it does not require a copy of the value in order to operate - you always operate using the value directly because it is holding the current value rather than the next one.

Reply:
In theory, `++x` is faster than `x++` for the reason you mention. However, in practice, for simple types the optimizer will rearrange things to avoid the copy. Even for more complex types this is often the case.

Furthermore, just because a construct is preferred doesn't mean you need to go out of your way to use over an alternative. If there are good reasons for `x++`, like here, you can use it.